xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -421,6 +421,19 @@ const std::string CPVRClient::GetFriendlyName() const
   return Name();
 }
 
+std::string CPVRClient::GetInstanceName() const
+{
+
+  if (Addon()->SupportsInstanceSettings())
+  {
+    std::string instanceName;
+    Addon()->GetSettingString(ADDON_SETTING_INSTANCE_NAME_VALUE, instanceName, InstanceId());
+    if (!instanceName.empty())
+      return instanceName;
+  }
+  return "";
```

Comment:
The check for an empty string is not needed, as you return an empty string anyway in this case (line 434).

Reply:
I disagree:  The `return ""` is there primarily for when the outer condition `SupportsInstanceSettings` is false.

The inner condition, `instanceName.empty()`, tests the result of the `Addon()->GetSettingString` method.  If that method fails, we fall through to the `return ""`, otherwise we return the value obtained.

Reply:
Sorry, this code works as designed even if you simply remove the check.
If the call to `GetsettingString`return an empty string, `instancename` will be empty, but you can just do `return instanceName;`

=>
```c++
    std::string instanceName;
    Addon()->GetSettingString(ADDON_SETTING_INSTANCE_NAME_VALUE, instanceName, InstanceId());
    return instanceName;
```

Reply:
I will do that.

Reply:
Done.